backend/src/test/java/site/coduo/referencelink/service/CategoryServiceTest.java

```diff
@@ -0,0 +1,182 @@
+import site.coduo.referencelink.service.dto.CategoryUpdateResponse;
+import site.coduo.utils.CascadeCleaner;
+
+@SpringBootTest
+class CategoryServiceTest extends CascadeCleaner {
+
+    @Autowired
+    private CategoryService categoryService;
+
+    @Autowired
+    private CategoryRepository categoryRepository;
+
+    @Autowired
+    private ReferenceLinkRepository referenceLinkRepository;
+
+    @Autowired
+    private PairRoomRepository pairRoomRepository;
+
+    @AfterEach
+    void tearDown() {
+        deleteAllPairRoomCascade();
+    }
+
+    @Test
+    @DisplayName("페어룸 생성 후 카테고리를 저장한다.")
+    void save_category() {
+        //given
+        final PairRoom pairRoom = pairRoomRepository.save(INK_REDDDY_ROOM);
+        pairRoomRepository.save(pairRoom);
+
+        //when
+        final CategoryCreateResponse createdCategory = categoryService.createCategory(ACCESS_CODE.getValue(),
+                new CategoryCreateRequest("자바"));
+        final List<CategoryReadResponse> categories = categoryService.findAllByPairRoomAccessCode(
+                ACCESS_CODE.getValue());
+
+        //then
+        assertThat(categories.stream().anyMatch(
+                category -> category.id().equals(createdCategory.id()) &&
+                            category.value().equals(createdCategory.value())))
+                .isTrue();
+    }
+
+    @Test
+    @DisplayName("카테고리를 수정한다.")
+    void update_category() {
+        //given
+        final PairRoom pairRoom = pairRoomRepository.save(INK_REDDDY_ROOM);
+        pairRoomRepository.save(pairRoom);
+
+        final String categoryName = "자바";
+        final CategoryCreateResponse createdCategory = categoryService.createCategory(ACCESS_CODE.getValue(),
+                new CategoryCreateRequest(categoryName));
+
+        //when
+        final CategoryUpdateResponse updatedCategory = categoryService.updateCategoryName(ACCESS_CODE.getValue(),
+                new CategoryUpdateRequest(categoryName, "파이썬"));
+
+        //then
+        final List<CategoryReadResponse> categories = categoryService.findAllByPairRoomAccessCode(
+                ACCESS_CODE.getValue());
+        //then
+        assertThat(categories.stream().anyMatch(
+                category -> category.id().equals(createdCategory.id()) &&
+                            category.value().equals(updatedCategory.updatedCategoryName())))
+                .isTrue();
+    }
+
+    @Test
+    @DisplayName("페어룸에 중복된 카테고리가 있는 경우 저장에 실패한다.")
+    void fail_save_category() {
+        //given
+        final PairRoom pairRoom = pairRoomRepository.save(INK_REDDDY_ROOM);
+        pairRoomRepository.save(pairRoom);
+
+        final CategoryCreateRequest categoryCreateRequest = new CategoryCreateRequest("자바");
+        categoryService.createCategory(ACCESS_CODE.getValue(), categoryCreateRequest);
+
+        //when & then
+        assertThatThrownBy(() -> categoryService.createCategory(ACCESS_CODE.getValue(), categoryCreateRequest))
```

Comment:
[의견]
사소하지만, assertThatThrownBy와 assertThatCode는 함수형 인터페이스를 인자로 받아요. 그래서 저는 상태에 대한 검증이 아닌 행위에 대한 검증이라고 보는 편입니다.  근데!! 여기서 주의할 점은 내부에 별도 행위가 중첩(ACCESS_CODE.getValue())되어 있다면, 바람직한 테스트 코드라고 보기 어려울거 가탕요.
```suggestion
        final var foo = ACCESS_CODE.getValue();
        assertThatThrownBy(() -> categoryService.createCategory(foo, categoryCreateRequest))
```